src/blueprint/azext_blueprint/custom.py

```diff
@@ -81,84 +92,174 @@ def import_blueprint_with_artifacts(cmd,
-                     resource_scope=None,):
-    return client.delete(resource_scope=resource_scope,
-                         blueprint_name=blueprint_name)
-
-
-def get_blueprint(client,
-                  blueprint_name,
-                  management_group=None,
-                  subscription=None,
-                  resource_scope=None,):
-    return client.get(resource_scope=resource_scope,
-                      blueprint_name=blueprint_name)
-
-
-def list_blueprint(client,
-                   management_group=None,
-                   subscription=None,
-                   resource_scope=None,):
-    return client.list(resource_scope=resource_scope)
+def _blueprint_validator(cmd):
+    from azure.cli.core.commands.client_factory import get_subscription_id
+    args = cmd.ctx.args
+    args.resource_scope = '/providers/Microsoft.Management/managementGroups/{}'.format(
+        args.management_group
+    ) if has_value(args.management_group) else '/subscriptions/{}'.format(
+        has_value(args.subscription_id) if has_value(args.subscription_id) else get_subscription_id(cmd.cli_ctx))
+
+
+class BlueprintCreate(_BlueprintCreate):
+
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        from azure.cli.core.aaz import AAZStrArg
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.management_group = AAZStrArg(
+            options=['--management-group', '-m'],
+            arg_group="Resource_scope",
+            help="Use management group for the scope of the blueprint.")
+        args_schema.subscription_id = AAZStrArg(
+            options=['--subscription', '-s'],
+            arg_group="Resource_scope",
+            help="Use subscription for the scope of the blueprint. If --management-group is not specified, "
+                 "--subscription value or the default subscription will be used as the scope.")
+        args_schema.resource_scope._required = False
+        args_schema.resource_scope._registered = False
+        return args_schema
+
+    def _cli_arguments_loader(self):
+        from knack.arguments import CLICommandArgument
+        args = super()._cli_arguments_loader()
+        args.append(("_subscription", CLICommandArgument(dest="_subscription", options_list=["--___subscription"])))
+        return args
+
+    def pre_operations(self):
+        args = self.ctx.args
+        if not has_value(args.resource_scope):
+            _blueprint_validator(self)
+
+
+class BlueprintUpdate(_BlueprintUpdate):
+
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        from azure.cli.core.aaz import AAZStrArg
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.management_group = AAZStrArg(
+            options=['--management-group', '-m'],
+            arg_group="Resource_scope",
+            help="Use management group for the scope of the blueprint.",
+            nullable=True

```

Comment:
It's not required to add nullable for them. They are not using the request directly.